src-java/kilda-model/src/main/java/org/openkilda/model/FlowMirrorPath.java

```diff
@@ -0,0 +1,328 @@
+     */
+    public FlowMirrorPath(@NonNull FlowMirrorPath entityToClone, FlowMirrorPoints flowMirrorPoints) {
+        data = FlowMirrorPathCloner.INSTANCE.deepCopy(entityToClone.getData(), flowMirrorPoints);
+    }
+
+    @Builder
+    public FlowMirrorPath(@NonNull PathId pathId, @NonNull Switch srcSwitch, @NonNull Switch destSwitch,
+                          int destPort, int destVlan, FlowSegmentCookie cookie, long bandwidth, boolean ignoreBandwidth,
+                          FlowPathStatus status, List<PathSegment> segments, boolean destWithMultiTable) {
+        data = FlowMirrorPathDataImpl.builder().pathId(pathId).srcSwitch(srcSwitch)
+                .destSwitch(destSwitch).destPort(destPort).destVlan(destVlan).cookie(cookie).bandwidth(bandwidth)
+                .ignoreBandwidth(ignoreBandwidth).status(status).destWithMultiTable(destWithMultiTable)
+                .build();
+        // The reference is used to link path segments back to the mirror path. See {@link #setSegments(List)}.
+        ((FlowMirrorPathDataImpl) data).flowPath = this;
+
+        if (segments != null && !segments.isEmpty()) {
+            data.setSegments(segments);
+        }
+    }
+
+    public FlowMirrorPath(@NonNull FlowMirrorPath.FlowMirrorPathData data) {
+        this.data = data;
+    }
+
+    /**
+     * Checks whether the flow path goes through a single switch.
+     *
+     * @return true if source and destination switches are the same, otherwise false
+     */
+    public boolean isSingleSwitchPath() {
+        return getSrcSwitchId().equals(getDestSwitchId());
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        FlowMirrorPath that = (FlowMirrorPath) o;
+        return new EqualsBuilder()
+                .append(getBandwidth(), that.getBandwidth())
+                .append(isIgnoreBandwidth(), that.isIgnoreBandwidth())
+                .append(getPathId(), that.getPathId())
+                .append(getSrcSwitchId(), that.getSrcSwitchId())
+                .append(getDestSwitchId(), that.getDestSwitchId())
+                .append(getDestPort(), that.getDestPort())
+                .append(getCookie(), that.getCookie())
+                .append(getTimeCreate(), that.getTimeCreate())
+                .append(getTimeModify(), that.getTimeModify())
+                .append(getStatus(), that.getStatus())
+                .append(getSegments(), that.getSegments())
+                .append(isDestWithMultiTable(), that.isDestWithMultiTable())
+                .isEquals();
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getPathId(), getSrcSwitchId(), getDestSwitchId(), getDestPort(), getBandwidth(),
+                isIgnoreBandwidth(), getTimeCreate(), getTimeModify(), getStatus(), getSegments(),
+                isDestWithMultiTable());
+    }
```

Comment:
maybe just use @EqualsAndHashcode annotation?

Reply:
First of all, this was done by analogy with other models. And I don't know how `@EqualsAndHashCode` will behave with `@Delegate`.